src/Pages/Dashboard/Dashboard.jsx

```diff
@@ -1,66 +1,82 @@
+            error: null,
+            historydata: null
+        };
+    }
+
+    handleChange = (event, value) => {
+        this.setState({value: value});
     };
-  }
 
-  handleChange = (event, value) => {
-    this.setState({ value: value });
-  };
+    render() {
+        const {classes} = this.props;
+        const {value} = this.state;
+        const data = (this.props.data || []).slice();
 
-  render() {
-    const { classes } = this.props;
-    const { value } = this.state;
-    const data = (this.props.data || []).slice();
-    return (
-      <Fragment>
-        <Typography variant="h4" color="primary" gutterBottom>
-          {this.props.sensorName + " Dashboard"}
-        </Typography>
-        <MonitoringData data={data} />
-        <br />
-        <Tabs
-          value={value}
-          className={classes.tabs}
-          onChange={this.handleChange}
-          indicatorColor="primary"
-          textColor="primary"
-          centered
-        >
-          <Tab className={classes.tab} label="Data" />
-          <Tab className={classes.tab} label="History" />
-          <Tab className={classes.tab} label="Upload CSV" />
-        </Tabs>
-        <br />
-        <br />
-        {
-          [
-            <ChartContainer data={this.props.data} />,
-            <Paper></Paper>,
-            <UploadData
-              firebase={this.props.firebase}
-              sensorId={this.props.sensorId}
-            />
-          ][value]
-        }
-      </Fragment>
-    );
-  }
+        return (
+            <Fragment>
+                <Typography variant="h4" color="primary" gutterBottom>
+                    {this.props.sensorName + " Dashboard"}
+                </Typography>
+                <MonitoringData data={data}/>
+                <br/>
+                <Tabs
+                    value={value}
+                    className={classes.tabs}
+                    onChange={this.handleChange}
+                    indicatorColor="primary"
+                    textColor="primary"
+                    centered
+                >
+                    <Tab className={classes.tab} label="Data"/>
+                    <Tab className={classes.tab} label="History"/>
+                    <Tab className={classes.tab} label="Upload CSV"/>
+                </Tabs>
+                <br/>
+                <br/>
+                {
+                    [
+                        <ChartContainer data={this.props.data}/>,
+                        <Grid container spacing={3}>
+                            <Grid item xs={12}>
+                                <ShowHistoryGraph
+                                    state={this.state}
+                                    selected={this.props.selected}
+                                    changeRange={this.props.changeRange}
+                                />
+                            </Grid>
+                            <Grid item xs={24} sm={24} md={12}>
+                                <ChartContainer
```

Comment:
This could probably be a component within `ShowHistoryGraph`. You just have to pass `historydata` into it and pass it along. This way we keep them together and it's more clear what it's doing at first glance.